src/process/plumerise/CCPr_Plumerise_Mod.F90

```diff
@@ -0,0 +1,271 @@
+
+      ! INPUT/OUTPUT PARAMETERS
+      !------------------------
+      INTEGER,          INTENT(INOUT) :: RC
+
+      ! LOCAL VARIABLES
+      !----------------
+      INTEGER :: k ! Loop Counter
+
+      ! Error handling
+      !---------------
+      CHARACTER(LEN=255)    :: ErrMsg
+      CHARACTER(LEN=255)    :: ThisLoc
+
+      ! Initialize Error handling
+      !--------------------------
+      ErrMsg = ''
+      ThisLoc = ' -> at CCPR_Plumerise_INIT (in process/Plumerise/ccpr_Plumerise_mod.F90)'
+
+      ! Initialize
+      !-----------
+      if (Config%Plumerise_activate) then
+
+         ! Activate Plumerise Process
+         !----------------------
+         Plumerisestate%Activate = .true.
+
+      else
+
+         PlumeriseState%Activate = .false.
+
+      endif
+
+
+   END SUBROUTINE CCPR_Plumerise_INIT
+
+   !>
+   !! \brief Run the Plumerise scheme
+   !!
+   !! \param [IN] MetState The MetState object
+   !! \param [INOUT] DiagState The DiagState object
+   !! \param [INOUT] PlumeriseState The PlumeriseState object
+   !! \param [INOUT] ChemState The ChemState object
+   !! \param [OUT] RC Return code
+   !!
+   !! \ingroup catchem_plumerise_process
+   !!!>
+   SUBROUTINE CCPr_Plumerise_Run(PlumeriseState, RC, verbose_opt )
+
+      ! USE
+      USE ccpr_scheme_sofiev_mod, ONLY : CCPr_Sofiev_Plmrise
+      use CCPR_Scheme_Briggs_Mod, only : CCPr_Briggs_Plumerise
+
+      IMPLICIT NONE
+
+      ! INPUT PARAMETERS
+      !-----------------
+      ! TYPE(MetStateType),  INTENT(IN) :: MetState       ! MetState Instance
+      ! TYPE(GridStateType), INTENT(IN) :: GridState      ! GridState Instance
+      ! TYPE(EmisStateType), INTENT(IN) :: EmisState      ! EmisState Instance
+
+      ! INPUT/OUTPUT PARAMETERS
+      !------------------------
+      TYPE(PlumeriseStateType), INTENT(INOUT) :: PlumeriseState   ! PlumeriseState Instance
+      ! TYPE(ChemStateType), INTENT(INOUT) :: ChemState  ! ChemState Instance
+
+      ! OUTPUT PARAMETERS
+      !------------------
+      INTEGER, INTENT(OUT) :: RC                         ! Return Code
+
+      LOGICAL, OPTIONAL, INTENT(IN) :: verbose_opt
+
+      ! LOCAL VARIABLES
+      !----------------
+      CHARACTER(LEN=255) :: ErrMsg, thisLoc
+      Integer :: c ! Emission Categories
+      Integer :: s ! Emitted Species
+      Integer :: p ! Plumerise source length counter
+      Integer :: z ! Vertical counter
+      Logical :: verbose
+
+      REAL(fp) :: plmHGT                              ! Plumerise Height [m]
+      REAL(fp) :: EFRAC(GridState%number_of_levels)   ! Emission fraction
+      real(fp) :: ColEmis(GridState%number_of_levels) ! Column emission rate [kg/m2/s]
+
+      ! Initialize
+      !-----------
+      RC = CC_SUCCESS
+      errMsg = ''
+      thisLoc = ' -> at CCPr_Plumerise_Run (in process/Plumerise/ccpr_Plumerise_mod.F90)'
+
+      if (present(verbose_opt) .eqv. .true.) then
+         if (verbose_opt .eqv. .true.) then
+            verbose=.TRUE.
```

Comment:
Think it would look nicer with the subroutine arg being just `verbose`. Then the local variable could be `verbose_` or such.